components/lcd/esp_lcd_sh1107/esp_lcd_sh1107.c

```diff
@@ -0,0 +1,324 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <sys/cdefs.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_lcd_panel_interface.h"
+#include "esp_lcd_panel_io.h"
+#include "esp_lcd_panel_vendor.h"
+#include "esp_lcd_panel_ops.h"
+#include "esp_lcd_panel_commands.h"
+#include "esp_lcd_sh1107.h"
+#include "driver/gpio.h"
+#include "esp_log.h"
+#include "esp_check.h"
+
+static const char *TAG = "sh1107";
+
+#define LCD_SH1107_I2C_CMD  0X00
+#define LCD_SH1107_I2C_RAM  0X40
+
+static esp_err_t panel_sh1107_del(esp_lcd_panel_t *panel);
+static esp_err_t panel_sh1107_reset(esp_lcd_panel_t *panel);
+static esp_err_t panel_sh1107_init(esp_lcd_panel_t *panel);
+static esp_err_t panel_sh1107_draw_bitmap(esp_lcd_panel_t *panel, int x_start, int y_start, int x_end, int y_end, const void *color_data);
+static esp_err_t panel_sh1107_invert_color(esp_lcd_panel_t *panel, bool invert_color_data);
+static esp_err_t panel_sh1107_mirror(esp_lcd_panel_t *panel, bool mirror_x, bool mirror_y);
+static esp_err_t panel_sh1107_swap_xy(esp_lcd_panel_t *panel, bool swap_axes);
+static esp_err_t panel_sh1107_set_gap(esp_lcd_panel_t *panel, int x_gap, int y_gap);
+static esp_err_t panel_sh1107_disp_on_off(esp_lcd_panel_t *panel, bool off);
+
+typedef struct {
+    esp_lcd_panel_t base;
+    esp_lcd_panel_io_handle_t io;
+    int reset_gpio_num;
+    bool reset_level;
+    int x_gap;
+    int y_gap;
+    unsigned int bits_per_pixel;
+    uint16_t lcd_width;
+    uint16_t lcd_height;
+    bool swap_axes;
+} sh1107_panel_t;
+
+esp_err_t esp_lcd_new_panel_sh1107(const esp_lcd_panel_io_handle_t io, const esp_lcd_panel_dev_config_t *panel_dev_config, esp_lcd_panel_handle_t *ret_panel)
+{
+#if CONFIG_LCD_ENABLE_DEBUG_LOG
+    esp_log_level_set(TAG, ESP_LOG_DEBUG);
```

Comment:
Have you tested this scenario? in previous version of IDF you had to set global log level to 'debug' too...

Reply:
Sorry, it left here, when I copied from esp_lcd. Removed.